Mathlib/Algebra/MonoidAlgebra/Basic.lean

```diff
@@ -452,6 +451,17 @@ theorem single_mul_single [Mul G] {a₁ a₂ : G} {b₁ b₂ : k} :
     (sum_single_index (by rw [mul_zero, single_zero]))
 #align monoid_algebra.single_mul_single MonoidAlgebra.single_mul_single
 
+theorem single_commute_single [Mul G] {a₁ a₂ : G} {b₁ b₂ : k}
+    (ha : Commute a₁ a₂) (hb : Commute b₁ b₂) :
+    Commute (single a₁ b₁) (single a₂ b₂) :=
+  single_mul_single.trans <| congr_arg₂ single ha hb |>.trans single_mul_single.symm
+
+theorem single_commute [Mul G] {a : G} {b : k} (ha : ∀ a', Commute a a') (hb : ∀ b', Commute b b') :
+    ∀ f : MonoidAlgebra k G, Commute (single a b) f :=
```

Comment:
```suggestion
theorem isMulCentral_commute [Mul G] {a : G} {b : k} (ha : IsMulCentral a) (hb : IsMulCentral b) :
    IsMulCentral (single a b) :=
```
Annoyingly, this comes with associativity obligations, so maybe make the above a separate lemma and instead change this one to
```suggestion
theorem forall_commute_single [Mul G] {a : G} {b : k} (ha : ∀ a', Commute a a') (hb : ∀ b', Commute b b') :
    ∀ f : MonoidAlgebra k G, Commute (single a b) f :=
```

Reply:
The `center` version requires some prework that I'd rather leave to another PR; #6993 is blocked by this PR, and doesn't care about non-associative rings.

Reply:
The prework: #9053